packages/features/tasker/tasks/triggerNoShow/common.ts

```diff
@@ -0,0 +1,149 @@
+import dayjs from "@calcom/dayjs";
+import { sendGenericWebhookPayload } from "@calcom/features/webhooks/lib/sendPayload";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import type { TimeUnit } from "@calcom/prisma/enums";
+import { BookingStatus, WebhookTriggerEvents } from "@calcom/prisma/enums";
+
+import { getBooking } from "./getBooking";
+import { getMeetingSessionsFromRoomName } from "./getMeetingSessionsFromRoomName";
+import type { TWebhook, TTriggerNoShowPayloadSchema } from "./schema";
+import { ZSendNoShowWebhookPayloadSchema } from "./schema";
+
+export type Host = {
+  id: number;
+  email: string;
+};
+
+export type Booking = Awaited<ReturnType<typeof getBooking>>;
+type Webhook = TWebhook;
+export type Participants = TTriggerNoShowPayloadSchema["data"][number]["participants"];
+
+export function getHosts(booking: Booking): Host[] {
+  const hostMap = new Map<number, Host>();
+
+  const addHost = (id: number, email: string) => {
+    if (!hostMap.has(id)) {
+      hostMap.set(id, { id, email });
+    }
+  };
+
+  booking?.eventType?.hosts?.forEach((host) => addHost(host.userId, host.user.email));
+  booking?.eventType?.users?.forEach((user) => addHost(user.id, user.email));
+
+  // Add booking.user if not already included
+  if (booking?.user?.id && booking?.user?.email) {
+    addHost(booking.user.id, booking.user.email);
+  }
+
+  // Filter hosts to only include those who are also attendees
+  const attendeeEmails = new Set(booking.attendees?.map((attendee) => attendee.email));
+  const filteredHosts = Array.from(hostMap.values()).filter(
+    (host) => attendeeEmails.has(host.email) || host.id === booking.user?.id
+  );
+
+  return filteredHosts;
+}
+
+export function sendWebhookPayload(
+  webhook: Webhook,
+  triggerEvent: WebhookTriggerEvents,
+  booking: Booking,
+  maxStartTime: number,
+  hostEmail?: string
+): Promise<any> {
+  const maxStartTimeHumanReadable = dayjs.unix(maxStartTime).format("YYYY-MM-DD HH:mm:ss Z");
+
+  return sendGenericWebhookPayload({
+    secretKey: webhook.secret,
+    triggerEvent,
+    createdAt: new Date().toISOString(),
+    webhook,
+    data: {
+      bookingId: booking.id,
```

Comment:
I think it would be good to add another field for the user's email (the hostEmail) who didn't join. Now we only mention that in the message